### ISSUE TITLE
Question about the loss implementation

### ISSUE THREAD

**Issue by H3Art-q** [yifanzhang-pro/Matrix-SSL#4]
Hello, I am using the content of your work to build classification models for data in other fields, but I found that in the code implementation of the repo, in the `train` method part, you only used:

```python
mec_loss = (loss_func(p1, z2, lamda_inv) + loss_func(p2, z1, lamda_inv)) * 0.5 / args.m

mce_loss = (mce_loss_func(p2, z1, correlation=True, logE=False, lamda=args.mce_lambd, mu=args.mce_mu, order=args.mce_order, align_gamma=args.align_gamma)
            + mce_loss_func(p1, z2, correlation=True, logE=False, lamda=args.mce_lambd, mu=args.mce_mu, order=args.mce_order, align_gamma=args.align_gamma)
            + args.gamma * mce_loss_func(p2, z1, correlation=False, logE=False, lamda=args.mce_lambd, mu=args.mce_mu, order=args.mce_order, align_gamma=args.align_gamma)
            + args.gamma * mce_loss_func(p1, z2, correlation=False, logE=False, lamda=args.mce_lambd, mu=args.mce_mu, order=args.mce_order, align_gamma=args.align_gamma)
            ) * 0.5

loss = mce_loss

optimizer.zero_grad()
loss.backward()
optimizer.step()
```

Does this mean that **MEC loss** does not participate in the training framework of the entire model? Is this part of the code redundant?

**Comment by yifanzhang-pro** [comment]
Thanks for your attention, we will update the arxiv paper soon. In our paper, we have pointed out that our method contains matrix alignment loss and matrix uniformity loss, well the MEC loss part, is connected to the matrix uniformity loss, which corresponds to the first two terms of `mce_loss` quoted in your question.
